### PR TITLE
More item commands

### PR DESCRIPTION
**Pull request by kannibalox**

### CODE REVIEW

File: docs/include-cmd-items.rst [hunk 1]
```diff
@@ -211,7 +211,17 @@ When called within configuration methods or in a ``Ctrl-X`` prompt, the target i
     d.custom2…5
     d.custom2…5.set
 
-        **TODO**
+        .. code-block:: ini
+
+            d.custom = string ‹key› ≫ string ‹value›
+            d.custom.set = string ‹key›, string ‹value› ≫ 0
+            d.custom1 = string ‹key› ≫ string ‹value›
```
Comment (pyroscope) [inline docs/include-cmd-items.rst]: wrong signature

File: docs/include-cmd-items.rst [hunk 2]
```diff
@@ -211,7 +211,17 @@ When called within configuration methods or in a ``Ctrl-X`` prompt, the target i
...
+            d.custom1.set = string ‹key›, string ‹value› ≫ 0
+
+
+        Set and return custom values with arbitary keys. Note that ``d.custom1=`` is *not* the same as ``d.custom=1`` or ``d.custom=custom1``, and can only be accessed by its assigned commands.
```
Comment (pyroscope) [inline docs/include-cmd-items.rst]: ^^^ arbitary
vvv doens't

File: docs/include-cmd-items.rst [hunk 3]
```diff
@@ -281,11 +345,27 @@ When called within configuration methods or in a ``Ctrl-X`` prompt, the target i
...
     d.message
     d.message.set
+
+        .. code-block:: ini
+
+            d.message = ‹hash› ≫ string ‹message›
+            d.message.set = ‹hash›, string ‹message› ≫ 0
+
+
+        Used to store messages relating to the item, such as errors in communicating with the traker or a hash check failure.
```
Comment (pyroscope) [inline docs/include-cmd-items.rst]: traker

File: docs/include-cmd-items.rst [hunk 4]
```diff
@@ -317,7 +397,9 @@ When called within configuration methods or in a ``Ctrl-X`` prompt, the target i
 
     d.ratio
 
-        **TODO**
+        Returns the ratio of the torrent. This is the amount of uploaded data divided by the completed bytes in the items.
```
Comment (pyroscope) [inline docs/include-cmd-items.rst]: needs to mention the scale (×1000).

File: docs/include-cmd-items.rst [hunk 5]
```diff
@@ -329,14 +411,34 @@ When called within configuration methods or in a ``Ctrl-X`` prompt, the target i
...
+
     d.skip.rate
     d.skip.total
+
+        .. code-block:: ini
+
+            d.skip.rate = ‹hash› ≫ value ‹rate›
+            d.skip.total = ‹hash› ≫ value ‹total›
+
+        Skipped pieces are ones that are recieved from a peer, but aren't needed and are thus ignored. This
```
Comment (pyroscope) [inline docs/include-cmd-items.rst]: recieved

File: docs/include-cmd-items.rst [hunk 6]
```diff
@@ -368,7 +470,16 @@ When called within configuration methods or in a ``Ctrl-X`` prompt, the target i
     d.uploads_min
     d.uploads_min.set
 
-        **TODO**
+        .. code-block:: ini
+
+            d.uploads_max = ‹hash› ≫ value ‹max›
+            d.uploads_max.set = ‹hash›, value ‹max› ≫ 0
+            d.uploads_min = ‹hash› ≫ value ‹max›
```
Comment (pyroscope) [inline docs/include-cmd-items.rst]: s/max/min/

**Review by pyroscope — CHANGES_REQUESTED** [review]
some fixes. nice cross-links. ;)

**Comment by kannibalox** [comment]
Ugh I need to figure out a good way to spell check, too many false positives on the command names at the moment.

**Pull request merged**